client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -0,0 +1,351 @@
+	"""
+	homepage = 'https://github.com/securestate/king-phisher'
+
+	def initialize(self):
+		module = sys.modules['king_phisher.client.application']
+		self.temp = module.KingPhisherClientApplication.do_sftp_client_start
+		module.KingPhisherClientApplication.do_sftp_client_start = self.new_SFTP_signal
+		sys.modules['king_phisher.client.application'] = module
+		return True
+
+	def _cleanup(self):
+		module = sys.modules['king_phisher.client.application']
+		module.KingPhisherClientApplication.do_sftp_client_start = self.temp
+		sys.modules['king_phisher.client.application'] = module
+
+	def new_SFTP_signal(self):
+		if not self.application._ssh_forwarder:
+			config = self.application.config
+			server = config['server']
+			server = server.split(':')
+			username = config['server_username']
+			password = config['server_password']
+			self.application._create_ssh_forwarder(server, username, password)
+		connection = self.application._ssh_forwarder
+		ssh = connection.client
+		ftp = ssh.open_sftp()
+
+		target_file = os.path.splitext(__file__)[0]
+		builder = Gtk.Builder()
+		builder.add_from_file(target_file + '.ui')
+		self.window = builder.get_object('SFTPClientGUI.window')
+		manager = FileManager(builder, self.application, ftp)  #pylint: disable=unused-variable
+		self.window.show_all()
+
+class Logger(object):
+	def __init__(self, builder, *args, **kwargs):
+		self.builder = builder
+		self.scroll = self.builder.get_object('scrolledwindow2')
+		self.treeview_transfer = self.builder.get_object('SFTPClientGUI.treeview_transfer')
+		self.progress_bar = self.builder.get_object('SFTPClientGUI.progressbar')
+		self.label_file = self.builder.get_object('SFTPClientGUI.label')
+		col = Gtk.TreeViewColumn('Transfer')
+		colText = Gtk.CellRendererText()
+		col.pack_start(colText, True)
+		col.add_attribute(colText, "text", 0)
+		self.treeview_transfer.append_column(col)
+		self._tvmodel = Gtk.ListStore(str)
+		self.treeview_transfer.connect('size-allocate', self._treeview_changed)
+		self.treeview_transfer.set_model(self._tvmodel)
+		self.treeview_transfer.connect('button_press_event', self._signal_treeview_button_pressed)
+
+		self.popup_menu = Gtk.Menu.new()
+		menu_item = Gtk.MenuItem.new_with_label('Clear')
+		menu_item.connect('activate', self.clear_log)
+		self.popup_menu.append(menu_item)
+		self.popup_menu.show_all()
+
+	def _signal_treeview_button_pressed(self, _, event):
+		if event.button == 3:
+			self.popup_menu.popup(None, None, None, None, event.button, Gtk.get_current_event_time())
+			return True
+		return
+
+	def log_generic(self, string):
+		self._tvmodel.prepend((string,))
+
+	def clear_log(self, _):
+		self._tvmodel.clear()
+
+	def _treeview_changed(self, widget, event, data=None):
+		adj = self.scroll.get_vadjustment()
+		adj.set_value(0)
+
+	def log_transfer_upload(self, path, size, isFolder, error=False):
+		if error:
+			string = 'Error uploading ' + path
+			self._tvmodel.prepend((string,))
```

Comment:
All three of these `self._tvmodel.prepend(` calls should be consolidated at the bottom of this function. Just set string in the conditional blocks then prepend it at the end.